dll/cpl/sysdm/general.c

```diff
@@ -417,14 +419,330 @@ static VOID SetProcSpeed(HWND hwnd, HKEY hKey, LPTSTR Value, UINT uID)
+        HeapFree(GetProcessHeap(), 0, ls2);
+    }
+    return result;
+}
+
+/*
+ * Replaces full word with another shorter word
+ */
+static
+VOID wcsrep(
+    _Inout_ PWSTR pwStr,
+    _In_ PWSTR pwFind,
+    _In_ PWSTR pwReplace)
+{
+    PWSTR pwsStr, pwsFind, pwsReplace, pwsBuf = NULL;
+    SIZE_T lenStr;
+    SIZE_T lenFind;
+    SIZE_T lenReplace;
+
+    if (!pwStr || !pwFind || !pwReplace ||
+        wcslen(pwStr) == 0 ||
+        wcslen(pwFind) == 0 ||
+        wcslen(pwFind) < wcslen(pwReplace))
+    {
+        return;
+    }
+    lenStr = wcslen(pwStr) + 2 + 1;
+    lenFind = wcslen(pwFind) + 2 + 1;
+    lenReplace = wcslen(pwReplace) + 2 + 1;
+
+    pwsStr = HeapAlloc(GetProcessHeap(), 0, lenStr * sizeof(WCHAR));
+    if (!pwsStr)
+    {
+        return;
+    }
+    StringCchCopyW(pwsStr, lenStr, L" ");
+    StringCchCatW(pwsStr, lenStr, pwStr);
+    StringCchCatW(pwsStr, lenStr, L" ");
+
+    pwsFind = HeapAlloc(GetProcessHeap(), 0, lenFind * sizeof(WCHAR));
+    if (!pwsFind)
+    {
+        goto freeStr;
+    }
+    StringCchCopyW(pwsFind, lenFind, L" ");
+    StringCchCatW(pwsFind, lenFind, pwFind);
+    StringCchCatW(pwsFind, lenFind, L" ");
+
+    if (!(pwsBuf = wcsistr(pwsStr, pwsFind)))
+    {
+        goto freeFind;
+    }
+
+    pwsReplace = HeapAlloc(GetProcessHeap(), 0, lenReplace * sizeof(WCHAR));
+    if (!pwsReplace)
+    {
+        goto freeFind;
+    }
+    StringCchCopyW(pwsReplace, lenReplace, L" ");
+    StringCchCatW(pwsReplace, lenReplace, pwReplace);
+    StringCchCatW(pwsReplace, lenReplace, L" ");
+
+    do
+    {
+        // replace substring
+        memmove(pwsBuf, pwsReplace, (lenReplace - 1) * sizeof(WCHAR));
+        // shift characters
+        memmove(pwsBuf + lenReplace - (wcslen(pwReplace) > 0 ? 1 : 2), pwsBuf + lenFind - 1, (lenStr - lenFind - (pwsBuf - pwsStr) + 1) * sizeof(WCHAR));
+    }
+    while ((pwsBuf = wcsistr(pwsStr, pwsFind)) != NULL);
+
+    TrimDmiStringW(pwsStr);
+    StringCchCopyW(pwStr, wcslen(pwStr), pwsStr);
+
+    HeapFree(GetProcessHeap(), 0, pwsReplace);
+freeFind:
+    HeapFree(GetProcessHeap(), 0, pwsFind);
+freeStr:
+    HeapFree(GetProcessHeap(), 0, pwsStr);
+}
+
+static BOOL GetSystemName(PWSTR pBuf, SIZE_T cchBuf)
+{
+    PVOID SMBiosBuf;
+    PCHAR DmiStrings[ID_STRINGS_MAX] = { 0 };
+    WCHAR ven[512], dev[512];
+    UINT i;
+
+    SMBiosBuf = LoadSMBiosData(DmiStrings);
+    if (!SMBiosBuf)
+    {
+        return FALSE;
+    }
+
+    GetSMBiosStringW(DmiStrings[SYS_VENDOR], ven, _countof(ven), TRUE);
+    GetSMBiosStringW(DmiStrings[SYS_PRODUCT], dev, _countof(dev), TRUE);
+
+    if (wcslen(dev) == 0 ||
+        !wcscmp(dev, ven) ||
+        !wcscmp(dev, L"All Series") || // some ASUS boards
+        !wcscmp(ven, L"To be filled by O.E.M.") || // various boards
+        !wcscmp(dev, L"To be filled by O.E.M.") || // various boards
+        !wcscmp(ven, L"To Be Filled By O.E.M.") || // various boards
+        !wcscmp(dev, L"To Be Filled By O.E.M.") || // various boards
```

Comment:
use wcsicmp?

Reply:
Also, not sure how many more are going to be here, might think about adding them in a struct/ array/ whatever